### PR TITLE
BFD-1458: Reconcile FI_NUM mapping

### PR DESCRIPTION
**Pull request by JFU-NAVA-PBC**
**JIRA Ticket:**
[BFD-1458](https://jira.cms.gov/browse/BFD-1458)

**User Story or Bug Summary:**

BFD V2 add mapping of CCW FI_NUM (fiscal intermediary number) as FHIR identifier (in addition to existing mapping as code value)

Also patched corresponding V2 tests gold sample files to make tests pass.

---

### What Does This PR Do?

Replace me.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [ ] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [ ] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [ ] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by rdatoc — COMMENTED** [review]
Looks good

**Review by aschey-forpeople — DISMISSED** [review]

**Review by rdatoc — APPROVED** [review]
I walked thru Jim's code while he explained his changes for a better understanding of the PR and AC

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 1]
```diff
@@ -3182,6 +3182,8 @@ static void mapEobCommonGroupInpOutHHAHospiceSNF(
     if (fiscalIntermediaryNumber.isPresent()) {
       eob.addExtension(
           createExtensionCoding(eob, CcwCodebookVariable.FI_NUM, fiscalIntermediaryNumber));
+      eob.addExtension(
+          createExtensionIdentifier(CcwCodebookVariable.FI_NUM, fiscalIntermediaryNumber));
```
Comment (meliGuzman) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3186]: @JFU-NAVA-PBC V1 sets the FI_NUM only as an extension identifier but I see that for V2 here you're setting it in 2 spots. Is that fine? Wouldn't that be duplicative?

**Review by mjburling — CHANGES_REQUESTED** [review]
@JFU-NAVA-PBC The sample endpoint responses appear to be well-formatted json, but they're _marked_ compared to the existing entries... it looks like you updated these by hand? Could you instead generate with maven specifying `-DgenerateTestData=true` and commit the generated versions?

**Review by dondevun — DISMISSED** [review]

File: apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v1/coverageSearchByPatientId.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
   "total" : 4,
   "link" : [ {
     "relation" : "self",
-    "url" : "https://localhost:IGNORED_FIELD/v1/fhir/Coverage?_format=json&beneficiary=Patient%2F567834"
+    "url" : "https://localhost:IGNORED_FIELD1/v1/fhir/Coverage/?beneficiary=567834"
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v1/coverageSearchByPatientId.json:11]: Why did all of these change to `IGNORED_FIELD1`?

Comment (mjburling) [reply apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v1/coverageSearchByPatientId.json:11]: Hmm. I mentioned this and other discrepancies in the purportedly generated endpoint-responses to @meliGuzman, and she pointed me in the right direction. Without any changes here to `EndpointJsonComparatorBase`, we must assume that these are only _partially_ generated and manually updated to include these bespoke `IGNORED_FIELD1` values.

Along the same lines, is that what's occurring with the rest of the query string?

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java
Comment (mjburling) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:1]: Adjustments like this to our transformers must be paired with updates to the data dictionary. The current entry for  `FI Number` in the latest copy of the BFD data dictionary for V2 resources does not appear to reflect either the existing implementation nor the solution you've offered here.

As part of BFD-1458, can you work within the current BFD data dictionary framework to offer the appropriate documentation?

**Comment by JFU-NAVA-PBC** [comment]
@mjburling @dondevun as discussed during sprint planning, I'll scale the scope of the ticket as an investigation (will update the ticket), and leave this PR to be closed without merging.

**Comment by JFU-NAVA-PBC** [comment]
FYI, for the diffs' in the re-generated test gold files under /bfd-server/bfd-server-war, those diffs are observed event if you do the test data regen on master, so it is a separate issue.

**Pull request closed without merging**